docs/DeveloperGuide.md

```diff
@@ -2,37 +2,121 @@
+- To retrieve, parse each string using "," as its delimiter and convert it into its corresponding objects, then add all the entries to the object list (mealList, drinkList, exerciseList)
 
 
 ## Product scope
 ### Target user profile
-
-{Describe the target user profile}
+- Have a need to manage their dietary intake and exercise routines effectively.
+- Prefer desktop applications over other types of platforms.
+- Can type quickly and prefer typing over mouse interactions.
+- Are reasonably comfortable using command-line interface (CLI) applications.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+The fitness app aims to help users manage their dietary habits and exercise routines more efficiently compared to traditional GUI-driven apps. 
+By offering a streamlined interface optimized for keyboard input and CLI interactions, users can track their meals, drinks, and exercises swiftly, allowing them to focus more on their fitness and nutritional goals and less on navigating through complex user interfaces.
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Priority | As a ... | I want to ...          | So that I can ...                                      |
+|----------|----------|------------------------|--------------------------------------------------------|
+| ***      |new user| see usage instructions | refer to them when I forget how to use the application |
+| ***      |user| add a meal | track my daily calorie intake                          |
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+1. Should work on any mainstream OS (Linux, Windows, MacOS) as long as it has Java 11 or above installed.
+2. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
 
 ## Glossary
 
-* *glossary item* - Definition
+* *meal* - Any food consumed.
+* *drink* - Any beverage consumed.
+* *exercise* - Any physical activity performed.
+* *calories* - Measure of energy derived from food.
+* *carbohydrates* - Macronutrient providing energy.
+* *proteins* - Macronutrient essential for growth and repair.
+* *fat* - Macronutrient important for energy storage and insulation.
+* *sugar* - Simple carbohydrate often added to food for sweetness.
+* *fiber* - Indigestible plant material aiding digestion.
+* *water* - Essential liquid for hydration and bodily functions.
 
-## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+## Instructions for manual testing
```

Comment:
Proper sequence of the launch and shutdown is provided with simple description, making it easier for the user to navigate through